libbeat/common/transport/transport.go

```diff
@@ -0,0 +1,49 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package transport
+
+import (
+	"errors"
+	"net"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Dialer interface {
+	Dial(network, address string) (net.Conn, error)
+}
+
+type DialerFunc func(network, address string) (net.Conn, error)
+
+var (
+	ErrNotConnected = errors.New("client is not connected")
+
+	debugf = logp.MakeDebug("transport")
+)
+
+func (d DialerFunc) Dial(network, address string) (net.Conn, error) {
+	return d(network, address)
+}
+
+func Dial(c *Config, network, address string) (net.Conn, error) {
+	d, err := MakeDialer(c)
+	if err != nil {
+		return nil, err
+	}
+	return d.Dial(network, address)
+}
```

Comment:
we should keep the dialer functions/types together in one file.

Reply:
I'm going to address all feedback related to the transport refactoring in a separate PR that will resolve https://github.com/elastic/beats/issues/16728.

Reply:
Please see https://github.com/elastic/beats/pull/16734.

Reply:
If the file doesn't become to big we could merge them all into one, but I'm referening to configurable Dialer constructor that combines all these into a single dialers based on the configuration.